### PR TITLE
CHG: [droid] let the system handle volume

### PR DESCRIPTION
**Pull request by koying**
This sets the droid system volume as the master.
With android being built in TV's, we cannot capture the volume keys anymore, as they manage the TV volume (and CEC -> AVR).

So this listen from volume changes from the system and adjust the Kodi volume rather than the contrary.

/cc @Montellese @MartijnKaijser

### CODE REVIEW

**Comment by koying** [comment]
jenkins build this please

**Comment by koying** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
the hardware buttons on a phone/tablet now control the system volume which in turn Kodi adjust itself too?

**Comment by koying** [comment]
Yep

File: tools/android/packaging/xbmc/src/org/xbmc/kodi/XBMCSettingsContentObserver.java.in [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package com.semperpax.@APP_NAME_LC@;
```
Comment (wsnipex) [inline tools/android/packaging/xbmc/src/org/xbmc/kodi/XBMCSettingsContentObserver.java.in]: hmz :)

Comment (koying) [inline tools/android/packaging/xbmc/src/org/xbmc/kodi/XBMCSettingsContentObserver.java.in]: Hehe. No wonder it failed;)

**Comment by koying** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
if @Montellese is ok with code changes i'm fine with it as well. Haven't tried it though but sounds sane

**Comment by wsnipex** [comment]
good to go?
jenkins build this please

**Comment by wsnipex** [comment]
ah, I guess some squashing would be in order.

**Comment by koying** [comment]
Yep ;)
Now, I'm confused whether we can still put in changes/features or we are already in freeze.
Or are we in "ok, but no api break" mode?

**Comment by MartijnKaijser** [comment]
no API break

**Comment by Montellese** [comment]
Looks good apart from the minors.

**Comment by koying** [comment]
jenkins build and merge

**Comment by uNiversaI** [comment]
@koying its 100% confirmed that this change, breaks the warning kay toast  from showing when pressing volume up or down on remote.

Just compiled master with this PR reverted and its back to working as intended.

**Comment by koying** [comment]
Well, yeah, it surely breaks the warning as ACTION_VOLAMP_UP and ACTION_VOLAMP_DOWN are never called ;)
Once again: Is there an actual issue at hand?

**Comment by uNiversaI** [comment]
I take it from that comment you aren’t going to fix it the issue caused by this. tbh the kay toast pop up for the warning serves a good purpose, else it wouldn’t be there in first place.

**Comment by koying** [comment]
The check is not needed because I force the volume to 100% on amlogic, which, afak, is the only chipset actually requiring it. If there are other chipsets requiring, i'll just force it the same way.

Now, I take from your response that it's purely theoretical, so I'll wait for actual bug reports, if any.

BTW, ```the warning serves a good purpose, else it wouldn’t be there in first place``` is not an approach I rely on. The Kodi codebase is 10 yrs old, with dozens of different committers on a dozen platform, so the risk of dead or unnecessary code is not quite zero ;)

**Comment by uNiversaI** [comment]
Well, im on x86 platform not droid this is broken for all platforms that rely on it now.

**Comment by koying** [comment]
Gee... So you're on Linux or Windows x86, not droid, and the kai does not show for you, right?
Would have been easier to say that right away

**Comment by koying** [comment]
Tested on win32 nightly and works ok.
Trac: http://trac.kodi.tv/ticket/15925

**Comment by da-anda** [comment]
I installed yesterdays nightly on my WETEK box running cyanogenMod and noticed some issues with this I believe. I had passthrough enabled and was watching a movie with DTS audio. Passthrough itself was fine until I tried to change the volume. Android applied the system volume change even that I was in passthrough and after that I only heard some painful noise until I changed the volume to 100% again.
Also, I think the volume was not restored correctly to my previous level after playback - I at least had to turn the volume down several times for other stuff.

**Pull request merged**